plugin/src/main/java/org/opensearch/ml/rest/RestMLPredictionAction.java

```diff
@@ -143,9 +149,10 @@ public RestChannelConsumer prepareRequest(RestRequest request, NodeClient client
     MLPredictionTaskRequest getRequest(String modelId, String modelType, String userAlgorithm, RestRequest request) throws IOException {
         String tenantId = getTenantID(mlFeatureEnabledSetting.isMultiTenancyEnabled(), request);
         ActionType actionType = ActionType.from(getActionTypeFromRestRequest(request));
-        if (FunctionName.REMOTE.name().equals(modelType) && !mlFeatureEnabledSetting.isRemoteInferenceEnabled()) {
+        if (modelType != null && FunctionName.REMOTE.name().equals(modelType) && !mlFeatureEnabledSetting.isRemoteInferenceEnabled()) {
```

Comment:
Adding null check for modelType here won't make any difference, since modelType is inside equals()